x-pack/plugins/data_visualizer/public/application/common/components/document_count_content/total_count_header.tsx

```diff
@@ -5,30 +5,58 @@
  * 2.0.
  */
 
-import { EuiFlexItem, EuiText } from '@elastic/eui';
+import { EuiFlexItem, EuiText, EuiLoadingSpinner, EuiIconTip } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n-react';
 import React from 'react';
+import { i18n } from '@kbn/i18n';
 
-export const TotalCountHeader = ({ totalCount }: { totalCount: number }) => {
+const SIGFIGS_IF_ROUNDING = 3; // Number of sigfigs to use for values < 10
+
+export const TotalCountHeader = ({
+  totalCount,
+  approximate,
+  loading,
+}: {
+  totalCount: number;
+  loading?: boolean;
+  approximate?: boolean;
+}) => {
   return (
-    <EuiFlexItem>
-      <EuiText size="s" data-test-subj="dataVisualizerTotalDocCountHeader">
+    <EuiFlexItem grow={false} style={{ flexDirection: 'row' }}>
+      <EuiText size="s" data-test-subj="dataVisualizerTotalDocCountHeader" textAlign="center">
         <FormattedMessage
           id="xpack.dataVisualizer.searchPanel.totalDocCountLabel"
-          defaultMessage="Total documents: {strongTotalCount}"
+          defaultMessage="Total documents: {prepend}{strongTotalCount}"
           values={{
-            strongTotalCount: (
+            prepend: !loading && approximate ? '~' : '',
+            strongTotalCount: loading ? (
+              <EuiLoadingSpinner size="s" />
+            ) : (
               <strong data-test-subj="dataVisualizerTotalDocCount">
                 <FormattedMessage
                   id="xpack.dataVisualizer.searchPanel.totalDocCountNumber"
                   defaultMessage="{totalCount, plural, one {#} other {#}}"
-                  values={{ totalCount }}
+                  values={{
+                    totalCount: approximate
+                      ? totalCount.toPrecision(SIGFIGS_IF_ROUNDING)
+                      : totalCount,
+                  }}
                 />
               </strong>
             ),
           }}
         />
       </EuiText>
+      {approximate ? (
+        <EuiIconTip
+          content={i18n.translate('xpack.dataVisualizer.searchPanel.randomSamplerMessage', {
+            defaultMessage:
+              'Random sampler is being used for the total document count and the chart. Values shown are estimated.',
```

Comment:
To align with my other suggestions and to front-load the most important info:
```suggestion
              'Approximate values are shown in the total document count and chart, which use random sampler aggregations.',
```

Reply:
Updated here https://github.com/elastic/kibana/pull/136150/commits/5959f472a0344c980918307d74fe4902005a40ad